docs/routing.md

```diff
@@ -0,0 +1,248 @@
+    CheckRouteResult -->|done| IsPhaseHit{phase == hit}
+    CheckRouteResult -->|skip or next| CheckRouteResult
+    IsPhaseHit --> |yes| EndIsPhaseHit[matching done]
+    IsPhaseHit --> |no| IsPhaseMiss{phase == miss}
+    IsPhaseMiss --> |yes| Set404[set 404]
+    Set404 --> NextPhaseIsWhat{path found or phase == miss/error}
+    IsPhaseMiss --> |no| NextPhaseIsWhat
+    NextPhaseIsWhat --> |true - set next phase to hit| checkPhase
+    NextPhaseIsWhat --> |get next phase| getNextPhase --> checkPhase
+
+    getNextPhase --> NextPhaseIs{determine next phase}
+    NextPhaseIs --> |none| NextIsFS[filesystem]
+    NextPhaseIs --> |filesystem| NextIsRW[rewrite]
+    NextPhaseIs --> |rewrite| NextIsRS[resource]
+    NextPhaseIs --> |resource| NextIsMS[miss]
+    NextPhaseIs --> |all others| NextIsMS
+```
+
+### Checking Source Routes
+
+For every source route in a phase, we need to check for a match and process it accordingly.
+
+#### Source Matching
+
+The first step in this part of the process is checking the value for the route matches up with the current path. At the same time, this also extracts any dynamic parameters.
+
+Checking a route is a match involves verifying that the request method is the same, the correct fields are or aren't specified, and the current path name matches.
+
+If the route was not a match, we simply return that we want to skip it.
+
+#### Overrides
+
+In different phases, different things may be applied to our response object, or modify the current requests details. The first thing we check for is whether a source route wants to override the current headers and status code and, if so, reset them.
+
+#### Middleware
+
+Next up, any middleware is run. Middleware occurs during the `none` phase, at the very start of the routing process. The response headers from the middleware and taken and applied to the request and the response object, where appropriate.
+
+Additionally, it is important to check whether the middleware was a rewrite, so that the current path can be updated.
+
+If running the middleware failed, we have to bail out of routing and return an error to the client.
+
+#### Headers and Status Code
+
+Source routes are able to specify headers and status codes to apply to the response. These are updated internally so that they can be applied to the final response object.
+
+There are two different types of headers that the build output config can provide. There are important ones, and normal ones. Whether headers should be treated as important is denoted by the presence of an `important` property on a source route. We believe that this indicated that the headers should take precendence over all others on the final response object.
+
+#### Destination
+
+If a source route wants to rewrite the current path to point to a different file, it specifies a destination property. These are taken and the current path is updated, along with applying any search parameters we have collected.
```

Comment:
What do you mean with "these are taken"? with "these" you mean the destinations?

what about
```suggestion
If a source route wants to rewrite the current path to point to a different file, it specifies a destination property. We take such destination from (matching) source routes and replace the current path with them, along with applying any search parameters we might have collected.
```
?